src/pip/_internal/network/download.py

```diff
@@ -161,13 +161,13 @@ def __init__(
         progress_bar,  # type: str
     ):
         # type: (...) -> None
-        self._session = session
+        self.session = session
```

Comment:
I think if we pass `session` into `RequirementPreparer` that would be easier to refactor later (one class to change, not two), and easier to understand since the `Downloader` interface remains very small.

Reply:
I am a bit skeptical about this: `RequirementPreparer.__init__` is already taking 11 parameters and a Downloader having a session seems to be natural enough.  I don't have a really strong opinion though, if you really think it's a good idea I'll do it.

Reply:
I think it is a good idea. `RequirementPreparer.__init__` has 11 documented and explicit dependencies that are clear when looking at the class definition and the 2 places it is instantiated. If it works for 11, I think 12 would also be OK. A Downloader needing a session to do its work is reasonable, but to me nothing about "downloader" implies that it shares that session with others.

Reply:
> nothing about "downloader" implies that it shares that session with others.

Aha, I missed this point.  I've adapted the patch to use `RequirementPreparer.session`.